nixos/modules/services/security/bitwarden_rs/default.nix

```diff
@@ -23,10 +23,21 @@ let
       if value != null then [ "${nameToEnvVar name}=${if isBool value then boolToString value else toString value}" ] else []
     ) cfg.config))));
 
+  # TODO: support backends other than sqlite
+  supportedBackupBackends = [ "sqlite" ];
+  bitwarden_rs = pkgs.bitwarden_rs.override { inherit (cfg) dbBackend; };
 in {
   options.services.bitwarden_rs = with types; {
     enable = mkEnableOption "bitwarden_rs";
 
+    dbBackend = mkOption {
+      type = enum [ "sqlite" "mysql" "postgresql" ];
```

Comment:
I personally think that this is pretty misleading. When I see allowed values e.g. in a man-page, I'd expect all of them to work. So we should either update the description accordingly or (preferably) leave out `mysql` and `postgresql` for now.

Reply:
I agree, it would be best to simply the PR and just force the use of the sqlite backend for now. That way version 1.13 can already be used, and once people truly want to start different backends, proper attention can be put into it to make sure they are actually completely supported, like in the backup script and maybe other parts.

Reply:
@Ma27 @msteen Is the current implementation all right?

Reply:
I shall test the the changes once I am home, but I would prefer the files to be changed to not include the other databases and not make sqlite optional, but required, since for now we make it the required backend until further testing / requests are made.

Reply:
There seems to be a bug in the rust package build script because it adds the test flags after `--`, causing them not to be interpreted as flags, so the `--features` argument that it passed to it ends up being ignored. I will make a PR or a bug report for this later. For now it can be disabled by forcibly setting `checkPhase` to the empty string, because `dontCheck` does not seem to exist for rust packages, it has `doCheck` instead, but it seems to be ignored for the `checkPhase`, which is run regardless of what `doCheck` is set too. [Here is a gist](https://gist.github.com/msteen/e126394d9b70d553ca7aca0f3c8eb113) with how I would change the package for now, i.e. no need to change the module file, remove the patch, and update package definition as in the gist, which is the same as you had it without the bit about the databases, since we are defaulting to `sqlite` for now anyway, which does not need any additional packages. I have tested it on my desktop and it seems to work fine as far as I have tested it.